idx/bigtable/schema.go

```diff
@@ -0,0 +1,99 @@
+package bigtable
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/bigtable"
+	"github.com/grafana/metrictank/idx"
+	"github.com/raintank/schema"
+	log "github.com/sirupsen/logrus"
+)
+
+func SchemaToRow(def *schema.MetricDefinition) map[string][]byte {
+	row := map[string][]byte{
+		"Id":         []byte(def.Id.String()),
+		"OrgId":      make([]byte, 8),
+		"Name":       []byte(def.Name),
+		"Interval":   make([]byte, 8),
+		"Unit":       []byte(def.Unit),
+		"Mtype":      []byte(def.Mtype),
+		"Tags":       []byte(strings.Join(def.Tags, ";")),
+		"LastUpdate": make([]byte, 8),
+		"Partition":  make([]byte, 8),
+	}
+	binary.PutVarint(row["OrgId"], int64(def.OrgId))
+	binary.PutVarint(row["Interval"], int64(def.Interval))
+	binary.PutVarint(row["LastUpdate"], def.LastUpdate)
+	binary.PutVarint(row["Partition"], int64(def.Partition))
+	return row
+}
+
+func RowToSchema(row bigtable.Row, def *schema.MetricDefinition) error {
+	if def == nil {
+		return fmt.Errorf("cant write row to nill MetricDefinition")
+	}
+	columns, ok := row[COLUMN_FAMILY]
+	if !ok {
+		return fmt.Errorf("no columns in columnFamly %s", COLUMN_FAMILY)
+	}
+	*def = schema.MetricDefinition{}
+	var err error
+	var val int64
+	for _, col := range columns {
+		switch strings.SplitN(col.Column, ":", 2)[1] {
+		case "Id":
+			mkey, err := schema.MKeyFromString(string(col.Value))
+			if err != nil {
+				log.Errorf("bigtable-idx: load() could not parse ID %q: %s -> skipping", string(col.Value), err)
```

Comment:
why does this function have 3 different ways to handle errors?
* some errors returned to caller, at which point caller logs `"bigtable-idx: failed to marshal row to metricDef` as well as the returned error
* on this line, we log the error directly and skip reading the column, but continue reading other columns and return err=nil to the caller, this looks like a bug
* at the bottom of the loop, we Fatalf if we encounter an unknown column because "if we get bad data doesnt seem unreasonable"

seems we should treat all cases the same. i suggest to immediately return the error in all cases, and log.Fatalf in the caller